library/alloc/src/collections/binary_heap.rs

```diff
@@ -708,6 +710,93 @@ impl<T: Ord> BinaryHeap<T> {
         self.data.retain(f);
         self.rebuild();
     }
+
+    /// Returns `true` if the `BinaryHeap` contains the specified item.
+    ///
+    /// The item may be any borrowed form of the heap's item type, but
+    /// `PartialEq` on the borrowed form *must* match that for the item type.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// #![feature(binary_heap_contains_remove)]
+    /// use std::collections::BinaryHeap;
+    ///
+    /// let heap = BinaryHeap::from(vec![1, 3, 5, 7, 9]);
```

Comment:
Maybe the example could have a `BinaryHeap` of `String` and then call `contains("foo")` to demonstrate the `Borrow`-mechanism?